src/mumble/AudioInput.cpp

```diff
@@ -815,13 +816,30 @@ void AudioInput::encodeAudioFrame() {
 
 			if (g.s.iaeIdleAction == Settings::Deafen && !g.s.bDeaf) {
 				tIdle.restart();
+				if (g.s.bUndoIdleActionUponActivity) {
+					activityState = ActivityStateIdle;
+				}
 				emit doDeaf();
 			} else if (g.s.iaeIdleAction == Settings::Mute && !g.s.bMute) {
 				tIdle.restart();
+				if (g.s.bUndoIdleActionUponActivity) {
+					activityState = ActivityStateIdle;
+				}
 				emit doMute();
 			}
 		}
 
+		if (g.s.iaeIdleAction != Settings::Nothing && g.s.bUndoIdleActionUponActivity) {
```

Comment:
Can we change this up to ensure the value of activityState is always consistent? I'd like activityState to always be truthful, even if the new setting is not turned on. We might use it for other things in the future.

Something like this:

	if (activityState == ActivityStateExitedIdle) {
		activityState = ActivityStateActive;
		if (g.s.iaeIdleAction != Settings::Nothing && g.s.bUndoIdleActionUponActivity) {
				if (g.s.iaeIdleAction == Settings::Deafen && g.s.bDeaf) {
					emit doDeaf();
				} else if (g.s.iaeIdleAction == Settings::Mute && g.s.bMute) {
					emit doMute();
				}
			}
		}
	}

(also, please fix the minor coding style issue: `if(` -- missing space between `if` and `(`)